device/transport/mqtt/test/_mqtt_receiver_test.js

```diff
@@ -126,152 +91,129 @@ describe('MqttReceiver', function () {
         receiver.on('method_Reboot', function () { });
         receiver.on('message', function () { });
         receiver.on('message', function () { });
 
         // assert
-        assert.isTrue(onSpy.calledOnce,
-          'mqttClient.on was called more than once');
-        assert.isTrue(onSpy.calledWith('message'),
-          'mqttClient.on was not called for "message" event');
-        assert.isTrue(subscribeStub.calledTwice,
-          'mqttClient.subscribe was not called twice');
-        assert.isTrue(subscribeStub.firstCall.calledWith('$iothub/methods/POST/#', { qos: 0}),
-          'first call to mqttClient.subscribe was not with topic $iothub/methods/POST/#');
-        assert.isTrue(subscribeStub.secondCall.calledWith('topic_message'),
-          'second call to mqttClient.subscribe was not with topic topic_message');
+        assert.isTrue(fakeMqttBase.on.calledTwice,
+        'fakeMqttBase.on was not called twice (error + message)');
+        assert.isTrue(fakeMqttBase.on.calledWith('error'),
+        'fakeMqttBase.on was not called for "error" event');
+        assert.isTrue(fakeMqttBase.on.calledWith('message'),
+          'fakeMqttBase.on was not called for "message" event');
+        assert.isTrue(fakeMqttBase.subscribe.calledTwice,
+          'fakeMqttBase.subscribe was not called twice');
+        assert.isTrue(fakeMqttBase.subscribe.firstCall.calledWith('$iothub/methods/POST/#', { qos: 0}),
+          'first call to fakeMqttBase.subscribe was not with topic $iothub/methods/POST/#');
+        assert.isTrue(fakeMqttBase.subscribe.secondCall.calledWith(receiver._topicMessageSubscribe),
+          'second call to fakeMqttBase.subscribe was not with topic _topicMessageSubscribe');
       });
 
       // Tests_Tests_SRS_NODE_DEVICE_MQTT_RECEIVER_13_002: [ When a listener is added for the method event, the topic should be subscribed to. ]*/
       it('does not subscribe twice to the same topic for multiple event registrations', function () {
-        // setup
-        var mqttClient = new FakeMqttClient();
-        var onSpy = sinon.spy(mqttClient, 'on');
-        var subscribeStub = sinon
-              .stub(mqttClient, 'subscribe');
-
-        // stub should call 'done' with no error
-        subscribeStub.onCall(0).callsArgWithAsync(2, null);
-        subscribeStub.onCall(1).callsArgWith(2, null);
-
-        // test
-        var receiver = new MqttReceiver(mqttClient, 'topic_message');
+        var receiver = new Mqtt(fakeConfig, fakeMqttBase);
         receiver.on('method_UpdateFirmware', function () { });
         receiver.on('method_Reboot', function () { });
         receiver.on('message', function () { });
         receiver.on('message', function () { });
 
         // assert
-        assert.isTrue(onSpy.calledOnce,
-          'mqttClient.on was called more than once');
-        assert.isTrue(onSpy.calledWith('message'),
+        assert.isTrue(fakeMqttBase.on.calledTwice,
```

Comment:
>fakeMqttBase.on.calledTwice [](start = 22, length = 27)

This seems less robust than the original.

Reply:
why? we're actually testing more stuff than before. 

---
In reply to: [141923278](https://github.com/Azure/azure-iot-sdk-node/pull/96#discussion_r141923278) [](ancestors = 141923278)